service/r2/kv/store.go

```
@@ -73,6 +73,15 @@ func (s *store) FetchNamespaces() (*rules.NamespacesView, error) {
 	}, nil
 }
 
+func (s *store) ValidateRuleSet(rs *rules.RuleSetSnapshot) error {
+	validator := s.opts.Validator()
+	// No validator is set so by default the ruleSetSnapshot is valid
+	if validator == nil {
+		return nil
```

Perhaps we should return an error here instead? If we accidentally forget setting the validator in the options, and someone calls the validation endpoint, we should return an error saying the validator is not set instead of returning ok?